plugin/v3/plugin.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto3";
+package cloudquery.plugin.v3;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/cloudquery/plugin-pb-go/pb/plugin/v3;plugin";
+
+service Plugin {
+  // Get the name of the plugin
+  rpc GetName(GetName.Request) returns (GetName.Response);
+  // Get the current version of the plugin
+  rpc GetVersion(GetVersion.Request) returns (GetVersion.Response);
+  // Configure the plugin with the given credentials and mode
+  rpc Init(Init.Request) returns (Init.Response);
+  // Get all tables the source plugin supports. Must be called after Init
+  rpc GetTables(GetTables.Request) returns (GetTables.Response);
+  // Start the sync the source plugin
+  rpc Sync(Sync.Request) returns (stream Sync.Response);
+  // Write resources
+  rpc Write(stream Write.Request) returns (Write.Response);
+  // Send signal to flush and close open connections
+  rpc Close(Close.Request) returns (Close.Response);
+}
+
+enum REGISTRY {
+  REGISTRY_GITHUB = 0;
+  REGISTRY_GRPC = 1;
+  REGISTRY_LOCAL = 2;
+}
+
+enum SCHEDULER {
+  SCHEDULER_DFS = 0;
+  SCHEDULER_ROUND_ROBIN = 1;
+}
+
+enum Operator {
+  EQUAL = 0;
+  LESS_THAN = 1;
+}
+
+enum PK_MODE {
+  DEFAULT = 0;
+  CQ_ID_ONLY = 1;
+}
```

Comment:
```suggestion
enum Registry {
  REGISTRY_UNSPECIFIED = 0;
  REGISTRY_GITHUB = 1;
  REGISTRY_GRPC = 2;
  REGISTRY_LOCAL = 3;
}

enum Scheduler {
  SCHEDULER_UNSPECIFIED = 0;
  SCHEDULER_DFS = 1;
  SCHEDULER_ROUND_ROBIN = 2;
}

enum PKMode {
  PK_MODE_UNSPECIFIED = 0;
  PK_MODE_DEFAULT = 0;
  PK_MODE_CQ_ID_ONLY = 1;
}
```

Reply:
Couple of suggestions:
 - Use Pascal case for enum names (follows the style guide: https://protobuf.dev/programming-guides/style/#enums)
 - Add an "_UNSPECIFIED" as the first entry in each case so it's possible to distinguish unset values
 - Removes unused enums

Reply:
addressed.